src/server/modules/api/admin-v1/__tests__/AdminApiV1Controller.test.ts

```diff
@@ -0,0 +1,250 @@
+import moment from 'moment'
+import httpMocks from 'node-mocks-http'
+import { ValidationError } from 'sequelize'
+import { createRequestWithUser } from '../../../../../../test/server/api/util'
+import { UrlV1Mapper } from '../../../../mappers/UrlV1Mapper'
+import { AlreadyExistsError, NotFoundError } from '../../../../util/error'
+import { AdminApiV1Controller } from '../AdminApiV1Controller'
+
+const urlManagementService = {
+  createUrl: jest.fn(),
+  updateUrl: jest.fn(),
+  changeOwnership: jest.fn(),
+  getUrlsWithConditions: jest.fn(),
+  bulkCreate: jest.fn(),
+}
+
+const userRepository = {
+  findById: jest.fn(),
+  findByEmail: jest.fn(),
+  findOrCreateWithEmail: jest.fn(),
+  findOneUrlForUser: jest.fn(),
+  findUserByUrl: jest.fn(),
+  findUrlsForUser: jest.fn(),
+  saveApiKeyHash: jest.fn(),
+  findUserByApiKey: jest.fn(),
+  hasApiKey: jest.fn(),
+}
+
+const urlV1Mapper = new UrlV1Mapper()
+
+const controller = new AdminApiV1Controller(
+  userRepository,
+  urlManagementService,
+  urlV1Mapper,
+)
+
+/**
+ * Unit tests for Admin API v1 controller.
+ */
+describe('AdminApiV1Controller', () => {
+  describe('createUrl', () => {
+    it('create, sanitize and transfer link to target email for admin API', async () => {
```

Comment:
nit: should the test descriptions on lines 42 and 106 be swapped? seems like 42 is for the same user, whereas 106 is for changing ownership to a different user

Reply:
oops, thanks for noticing this, updated in [13e92a9](https://github.com/opengovsg/GoGovSG/pull/2213/commits/13e92a9131b430be78a53f85551fc6d854cf735b)